### PR TITLE
Bypass cloudtrail s3 bucket and support TF provider AWS 4 

### PR DESCRIPTION
**Pull request by MonolithProjects**

### CODE REVIEW

File: kms.tf [hunk 1]
```diff
@@ -7,6 +7,14 @@ data "aws_caller_identity" "kms" {
 }
 
 data "aws_iam_policy_document" "trail_kms" {
+  #checkov:skip=CKV_AWS_1
+  #checkov:skip=CKV_AWS_41
```
Comment (dojci) [inline kms.tf]: This finding means "Ensure AWS access keys and secrets are not hard coded in infrastructure" and it's used only in provider, it has nothing to do with IAM, please remove this suppress finding.

File: kms.tf [hunk 2]
```diff
@@ -7,6 +7,14 @@ data "aws_caller_identity" "kms" {
 }
 
 data "aws_iam_policy_document" "trail_kms" {
```
Comment (dojci) [inline kms.tf:9]: We should create an issue for proper security findings validation and follow the least privileges principle. In issue, we can mention best practices for [AWS KMS key cloudtrail configuration](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/create-kms-key-policy-for-cloudtrail.html)

**Review by dojci — CHANGES_REQUESTED** [review]

File: kms.tf (hunk 2 above)
Comment (MonolithProjects) [reply kms.tf:9]: agree. Will add it to my backlog...

**Review by dojci — APPROVED** [review]
Please drop a note to the release comments on what happens when someone starts using this new version and how to prevent possible data loss.

**Pull request merged**